### DISCUSSION TITLE
Prefect live debugging

### DISCUSSION THREAD

**Discussion by Mdanic75** [PrefectHQ/prefect#14710] [Q&A]
Hi Prefect Team,

I've been working with Prefect for orchestrating my data workflows, and it's been a fantastic experience so far. However, I often find myself in need of debugging the code as the flows are running. 

Is there a built-in feature in Prefect that allows for live debugging, or is there a recommended approach to achieve this? Any tips or tools that you can suggest would be greatly appreciated.

Thank you for your support and for the great tool!

Best regards,
Daniel

**Comment by cicdw** [comment]
Hi @Mdanic75 and thank you! 

You can achieve this when executing locally for sure: if you set up your prefect profile to point to whatever API you are using and execute a flow interactively, breakpoints will be respected:

```python
# example_flow.py

from prefect import flow

@flow(name="Breakpoint example")
def my_flow(name: str, num: int):
    ### some code
    breakpoint() 
    ### more code
```

and then executing this flow in your favorite interpreter will both communicate with Prefect's API and respect your break points:

```python
>> from example_flow import my_flow
>> my_flow("chris", 42)
## breakpoint will open as normal
```

Is that the sort of live debugging you were looking for?
